byterun/minor_gc.c

```diff
@@ -170,6 +176,56 @@ void caml_set_minor_heap_size (asize_t bsz)
   reset_table ((struct generic_table *) &caml_custom_table);
 }
 
+mlsize_t caml_distance_to_closure_environment(value closure)
+{
+#ifndef NATIVE_CODE_AND_NO_NAKED_POINTERS
+  return 0;
+#else
+  int finished = 0;
+  value* start;
+  value* pos;
+  mlsize_t skip;
+
+  CAMLassert(Is_block(closure));
+  CAMLassert(Tag_val(closure) == Closure_tag);
+  CAMLassert(Wosize_val(closure) >= 2);
+
+  start = (value*) closure;
+  pos = start;
+
+  while (!finished) {
```

Comment:
You could refactor this loop to avoid testing `finished` twice, either as a `for` loop, or as a `while(1)` loop with `if (finished) break;` in the right place. I'd find the latter more readable, but it's probably a matter of taste.